core/capabilities/remote/target/request/client_request.go

```diff
@@ -56,9 +58,14 @@ func NewClientRequest(ctx context.Context, lggr logger.Logger, req commoncap.Cap
 	lggr.Debugw("sending request to peers", "execID", req.Metadata.WorkflowExecutionID, "schedule", peerIDToTransmissionDelay)
 
 	responseReceived := make(map[p2ptypes.PeerID]bool)
+
+	ctxWithCancel, cancelFn := context.WithCancel(ctx)
+	wg := &sync.WaitGroup{}
 	for peerID, delay := range peerIDToTransmissionDelay {
 		responseReceived[peerID] = false
+		wg.Add(1)
 		go func(peerID ragep2ptypes.PeerID, delay time.Duration) {
```

Comment:
Let's pass `ctx` here so the name collision avoidance doesn't leak around:
```suggestion
		go func(ctx context.Context, peerID ragep2ptypes.PeerID, delay time.Duration) {
```